spring-shell-core/src/main/java/org/springframework/shell/Shell.java

```diff
@@ -264,4 +284,24 @@ private String findLongestCommand(String prefix) {
 		return "".equals(result) ? null : result;
 	}
 
+	private static class ParameterToResolver implements Comparable<ParameterToResolver> {
```

Comment:
I don't understand the need for that new tuple class. Why not just sort the resolvers (see my comment above) and continue to return the first that matches? Why put the parameter index in the picture? There is something I'm not understanding here, let's discuss tomorrow on the call if that helps